app/src/main/java/com/sdsmdg/hareshkh/lectureassignment/MovieAdapter.java

```diff
@@ -0,0 +1,172 @@
+package com.sdsmdg.hareshkh.lectureassignment;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.List;
+
+import com.bumptech.glide.Glide;
+
+/**
+ * Created by Jaskirat on 03-02-2017.
+ */
+
+public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.MyViewHolder>
+{
+    private Context mContext;
+    private List<Movie> movieList;
+    private String line;
+
+    public class MyViewHolder extends RecyclerView.ViewHolder
+    {
+        public TextView genre, name, year;
+        public ImageView thumbnail;
+
+        public MyViewHolder(View view)
+        {
+            super(view);
+            name = (TextView) view.findViewById(R.id.title);
+            year = (TextView) view.findViewById(R.id.yr);
+            genre= (TextView) view.findViewById(R.id.gen);
+            thumbnail = (ImageView) view.findViewById(R.id.thumbnail);
+        }
+    }
+
+    public MovieAdapter(Context mContext, List<Movie> movieList) {
+        this.mContext = mContext;
+        this.movieList = movieList;
+    }
+
+    @Override
+    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
+    {
+        View itemView = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.movie_card, parent, false);
+
+        return new MyViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(final MyViewHolder holder, int position)
+    {
+        Movie movie = movieList.get(position);
+        holder.name.setText(movie.getName());
+        holder.year.setText(movie.getYear());
+        holder.genre.setText(movie.getGenre());
+
+
+        Bitmap bitmap,scaled;
+        Drawable drawable;
+
+        switch(movie.getImg())
```

Comment:
You do not need a switch case here. You could just write the following replacing the whole switch case code.
drawable = ContextCompat.getDrawable(mContext, movie.getImg());
 +                                          bitmap = ((BitmapDrawable)drawable).getBitmap();
 +                                          scaled=Bitmap.createScaledBitmap(bitmap,200,200,false);
 +                                          holder.thumbnail.setImageBitmap(scaled);